pymongo/auth_aws.py

```diff
@@ -60,6 +68,9 @@ def _authenticate_aws(credentials, sock_info):
         raise ConfigurationError("MONGODB-AWS authentication requires MongoDB version 4.4 or later")
 
     try:
+        # Enable credential caching.
+        set_use_cached_credentials(True)
```

Comment:
Should this be called once at the module level instead?

Reply:
Done